packages/router-plugin/src/get-route-snapshot.ts

```diff
@@ -0,0 +1,47 @@
+import { Type } from '@angular/core';
+import { ActivatedRouteSnapshot } from '@angular/router';
+import { createSelector } from '@ngxs/store';
+
+import { RouterState, RouterStateModel } from './router.state';
+
+export function getRouteSnapshot<T>(component: Type<T>) {
```

Comment:
We could move this onto `RouterState` to make it more discoverable.

Reply:
It could be as `RouterState.getRouteSnapshot` or `RouterState.getSnapshot` or `RouterState.getComponentSnapshot`.